cachi2/core/package_managers/yarn/main.py

```diff
@@ -228,19 +232,14 @@ def _generate_environment_variables() -> list[EnvironmentVariable]:
 
 def _verify_corepack_yarn_version(expected_version: semver.Version, source_dir: RootedPath) -> None:
     """Verify that corepack installed the correct version of yarn by checking `yarn --version`."""
-    installed_yarn_version = run_yarn_cmd(
-        ["--version"], source_dir, env={"COREPACK_ENABLE_DOWNLOAD_PROMPT": "0"}
-    ).strip()
-    try:
-        if installed_yarn_version != expected_version:
-            raise PackageManagerError(
-                f"Cachi2 expected corepack to install yarn@{expected_version} but instead "
-                f"found yarn@{installed_yarn_version}."
-            )
-    except ValueError as exc:
+    installed_yarn_version = extract_yarn_version_from_env(
+        source_dir,
+        env={"COREPACK_ENABLE_DOWNLOAD_PROMPT": "0"},
```

Comment:
We used an explicit env because we were calling into the Yarn cmd executor wrapper directly. That is now abstracted by the new version parser helper and so the helper can hardcode this bit, no need to pass the env to a single-purpose helper as an argument :).

Reply:
Done.